src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoFinalizerCheck.java

```diff
@@ -25,9 +25,28 @@
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
 /**
- * Checks that no method having zero parameters is defined
- * using the name <em>finalize</em>.
+ * <p>
+ * Verifies there are no {@code finalize()} methods defined in a class.
+ * </p>
+ * <p>
+ * See
+ * <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()">
+ * Object.finalize()</a>
+ * </p>
+ * <p>
+ * Rationale: Finalizers are unpredictable, often dangerous, and generally unnecessary.
+ * Their use can cause erratic behavior, poor performance, and portability problems.
```

Comment:
Neither Java nor xdoc had this information, where did it come from?

Reply:
This check is sparsely documented. I added some information to make it look like his twin brother, `NoClone`.